core/src/main/java/com/radixdlt/environment/EventProcessorOnRunner.java

```diff
@@ -70,7 +70,8 @@
 import java.util.Optional;
 
 /**
- * An Event Processor registered to run on a runner.
+ * An Event Processor registered to run on a runner. Can be used to process both, local and remote
+ * events.
```

Comment:
same here, why does it need to handle remote events too?

Reply:
The issue lies in the way `EventProcessorOnDispatch` is used in `RemoteDispatcherProvider` (see Dispatchers.java):
```
    @Inject private Set<EventProcessorOnDispatch<?>> onDispatchProcessors;
```
The original code makes no differentiation between local and remote events when `onDispatch` is concerned. This inevitably pulls `EventProcessor` into the same issue because of the `EventProcessorOnDispatch.getProcessor` method.

So, the comment is correct as for the current state of the things. Separation of `EventProcessorOnDispatch` for remote events requires reworking of the modules and bindings, which I'd prefer not to do now.

Reply:
I don't think much rework is needed, check my commit on `feature/NODE-646-follow-up-for-NODE-427-local-events`

Reply:
Incorporated your changes with minor tweaks.